tests/test_app.py

```diff
@@ -77,3 +78,21 @@ def reqctx2():
     assert r.result == "GET"
 
     assert pytest.raises(RuntimeError, reqctx2.delay)
+
+
+def test_task_logger_propagation():
+    """Test log propagation of Celery task."""
+    app = Flask("myapp")
+    app.config.from_object(eager_conf)
+    celery = create_celery_app(app)
+    celery.log.setup_task_loggers()
+
+    @celery.task()
+    def logtask():
+        logger = get_task_logger(__name__)
+        while logger.name != 'root':
```

Comment:
Yes, that might be better to avoid getting into an infinite loop if something changes.